File: 파이썬/16428.py
a, b = map(int, input().split())

if a == 0:
    print(0)
    print(0)
else:
    c = abs(a)
    d = abs(b)

    ans1 = c // d
    ans2 = c % d
    
    if a > 0 and b > 0:
        print(ans1)
        print(ans2)
    elif a > 0 and b < 0:
        print(-ans1)
        print(ans2)
    elif a < 0 and b > 0:
        ans1 = -ans1
        ans2 = -ans2
        if ans2 < 0:
            ans1 = ans1 - 1
            ans2 = ans2 + b
        print(ans1)
        print(ans2)
    else:
        ans2 = -ans2
        if ans2 < 0:
            ans1 = ans1 + 1
            ans2 = ans2 - b
        print(ans1)
        print(ans2)

File: C++/16428.cpp
#include <iostream>
using namespace std;

int Abs(int n);

int main(void)
{
	int a, b, c, d, ans1, ans2;
	scanf("%d %d", &a, &b);

	if(a == 0){
		printf("0\n0");
		return 0;
	}

	c = Abs(a);
	d = Abs(b);

	ans1 = c / d;
	ans2 = c % d;

	if(a > 0 && b > 0)
		printf("%d\n%d", ans1, ans2);
	else if(a > 0 && b < 0)
		printf("%d\n%d", -ans1, ans2);
	else if(a < 0 && b > 0){
		if(ans2 > 0){
			ans1++;
			ans2 = -ans2 + d;
		}
		printf("%d\n%d", -ans1, ans2);
	}
	else{
		if(ans2 > 0){
			ans1++;
			ans2 = -ans2 + d;
		}
		printf("%d\n%d", ans1, ans2);
	}
}

int Abs(int n)
{
	if(n < 0)
		return -n;
	else
		return n;
}